src/Fixer/Phpdoc/PhpdocNoUselessInheritdocFixer.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Phpdoc;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Remove inheritdoc tags from classy that does not inherit.
+ *
+ * @author SpacePossum
+ */
+final class PhpdocNoUselessInheritdocFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // min. offset 4 as minimal candidate is @: <?php\n/** @inheritdoc */class min{}
+        for ($index = 1, $count = count($tokens) - 4; $index < $count; ++$index) {
+            if (!$tokens[$index]->isGivenKind(array(T_CLASS, T_INTERFACE))) {
+                continue;
+            }
+
+            $index = $this->fixClassy($tokens, $index);
```

Comment:
remove negation from condition, move this inside the condition, 2 lines less and no negation ;)